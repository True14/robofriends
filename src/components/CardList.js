import React, { memo } from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
    const cardComponent = robots.map((robot, index) => {
        return <Card key={index} id={robot.id} name={robot.name} email={robot.email} />
    })
    return (
        <div>
            {cardComponent}
        </div>
    )
}

export default memo(CardList);