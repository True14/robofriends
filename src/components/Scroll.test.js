import React from "react";
import { shallow } from "enzyme";
import Scroll from "./Scroll";
import CardList from "./CardList";
describe("Test the Scroll Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Scroll />);
  });
  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders child components properly", () => {
    const childrenTestWrapper = shallow(
      <Scroll>
        <CardList robots={[]} />
      </Scroll>
    );
    expect(
      childrenTestWrapper.containsMatchingElement(<CardList robots={[]} />)
    ).toEqual(true);
  });
});
