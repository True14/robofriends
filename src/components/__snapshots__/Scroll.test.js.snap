// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Scroll Component renders without crashing 1`] = `
<div
  style={
    Object {
      "border": "1px solid black",
      "height": "80vh",
      "overflowY": "scroll",
    }
  }
/>
`;
